cluster/addons/dns/kube2sky/kube2sky.go

```diff
@@ -87,8 +88,23 @@ func mutateEtcdOrDie(mutator func() error) {
 }
 
 func newEtcdClient() (client *etcd.Client) {
-	// TODO: take a flag for etcd server(s).
-	client = etcd.NewClient([]string{"http://127.0.0.1:4001"})
+	maxConnectRetries := 12
+	for maxConnectRetries > 0 {
+		if _, _, err := tools.GetEtcdVersion(*etcd_server); err != nil {
```

Comment:
It's the tools package under "pkg/tools". It contained a number of nice etcd tools, which I modified and reused.